src/clojure/flambo/data_frame.clj

```diff
@@ -0,0 +1,330 @@
+(ns flambo.data-frame
+  (:require [flambo.api :as f :refer [defsparkfn untuple]])
+  (:import [org.apache.spark.sql DataFrame Row Column GroupedData]))
+
+
+;; ## DataFrame
+;;
+(defn agg
+  "Aggregate the entire dataframe without groups."
+  ([^DataFrame df exprs]
+   (.agg df exprs)))
+
+(defn apply-column
+  "Returns Column based on column name"
+  [^DataFrame df ^String column-name]
+  (.apply df column-name))
+
+(defn as
+  "Returns a new dataframe with an alias"
+  [^DataFrame df ^String alias]
+  (.as df alias))
+
+(def cache (memfn cache))
+
+(defn col
+  "Return the column with this name"
+  [^DataFrame df ^String col-name]
+  (.col df col-name))
+
+(defn collect
```

Comment:
Seems like you could keep the parallelism with the Spark names, `(defn collect-as-list` or something like that.